### PR TITLE
Upload wizard expression files (including bundles)  (SCP-3657)

### PR DESCRIPTION
**Pull request by devonbush**
SCP-3657, SCP-3656.  This adds expression file upload to the new upload wizard, and also supports bundling.  the UX got a 'seems ok' nod from Jerome, so it might change a bit later, but nothing major.  

image

MANUAL TESTING:
1. Create a new study (make sure react_upload_wizard feature flag is on)
2. confirm you are taken to the raw counts upload
3. upload one or more raw counts files (including an mtx+barcodes+genes file)
4. upload one or more expression files (including an mtx+barcodes+genes file)
5. confirm parse jobs are launched and ingest succeeds/fails appropriately

### CODE REVIEW

**Review by eweitz — CHANGES_REQUESTED** [review]
Good stuff!  I noted many nits, but overall it's exciting to see this moving along quickly.

I found one blocker, which seems trivial to fix -- see comments beginning "Blocker:".

The PR title and description need content (including Jira ticket in title for auditability).

**Review by bistline — CHANGES_REQUESTED** [review]
I am unable to get the form overlay that prevents uploading processed matrices before raw counts to render.  Several of the constants in `ProcessedExpressionStep.js` are coming back `undefined`, and the state is not being set correctly.  I do have both raw counts feature flags turned on for my testing account, so I'm not sure what the issue is.

Modulo that, and Eric's comments, the rest looks good to me.

**Review by eweitz — APPROVED** [review]

**Comment by devonbush** [comment]
> Good stuff! I noted many nits, but overall it's exciting to see this moving along quickly.
> 
> I found one blocker, which seems trivial to fix -- see comments beginning "Blocker:".
> 
> The PR title and description need content (including Jira ticket in title for auditability).

Yes, thanks for the fast reviews -- I created this PR, and then got pulled into a bunch of meetings before I could add content -- should have opened in draft first :)

File: app/javascript/components/upload/ExpressionFileForm.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+          allowedFileTypes={isMtxFile ? FileTypeExtensions.mtx : FileTypeExtensions.plainText}/>
+        <div className="form-group">
+          <label>Matrix file type:</label><br/>
+          <label className="sublabel">
+            <input type="radio"
+              name={`rawCountsType-${file._id}`}
+              value="false"
+              checked={!isMtxFile}
+              onChange={e => updateFile(file._id, { file_type: 'Expression Matrix' })} />
+              &nbsp;Expression Matrix
```
[the remark this answers is not in the document]
Comment (eweitz) [reply app/javascript/components/upload/ExpressionFileForm.js:43]: Sweet!  Also, which of those 3 strings is logged?

**Review by bistline — APPROVED** [review]

**Pull request merged**